apollo-runtime/src/main/java/com/apollographql/apollo/ApolloPrefetch.java

```diff
@@ -1,23 +1,33 @@
 package com.apollographql.apollo;
 
-import java.io.IOException;
+import com.apollographql.apollo.exception.ApolloException;
+import com.apollographql.apollo.exception.ApolloHttpException;
+import com.apollographql.apollo.exception.ApolloNetworkException;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
 public interface ApolloPrefetch {
 
-  void execute() throws IOException;
+  void execute() throws ApolloException;
 
   @Nonnull ApolloPrefetch enqueue(@Nullable Callback callback);
 
   ApolloPrefetch clone();
 
   void cancel();
 
-  interface Callback {
-    void onSuccess();
+  abstract class Callback {
```

Comment:
Nice, like the option to use a single callback and inheritance, or the split methods.